src/in-view.js

```diff
@@ -17,40 +17,46 @@ const inView = () => {
     * each registry.
     */
     const interval = 100;
-    const triggers = ['scroll', 'resize', 'load'];
+    const triggers = ['scroll', 'resize'];
 
     /**
     * Maintain a hashmap of all registries, a history
     * of selectors to enumerate, and an options object.
     */
     let selectors = { history: [] };
-    let options   = { offset: {}, threshold: 0, test: inViewport };
+    let options   = { offset: {}, threshold: 0, test: inViewport, async: true };
 
     /**
     * Check each registry from selector history,
     * throttled to interval.
     */
     const check = throttle(() => {
         selectors.history.forEach(selector => {
-            selectors[selector].check();
+            const registry = selectors[selector];
+            if (options.async) {
+                let elements = [].slice.call(document.querySelectorAll(selector));
+                if (elements.length !== registry.elements.length) {
+                    registry.elements = elements;
+                }
```

Comment:
To me, it seems like unless we do a deep comparison, this isn't helpful because we could have a situation where...

```javascript
elements.length == 1 

// and

registry.elements.length == 1

// but the elements are actually different, and since we only check length,
// we don't set them to the registry.
```